plugins/Transitions/API.php

```diff
@@ -162,7 +162,10 @@ private function deriveIdAction($actionName, $actionType)
         switch ($actionType) {
             case 'url':
                 $originalActionName = $actionName;
-                $actionName = Common::unsanitizeInputValue($actionName);
+                if (is_array($actionName) && count($actionName) !== 1) {
+                    throw new Exception('NoDataForAction');
+                }
+                $actionName = (is_array($actionName) ? reset($actionName) : Common::unsanitizeInputValue($actionName));
```

Comment:
Is this only specifically relevant to the `url` actionType or would it be a valid sanitisation for the `title` as well (or in general), and worth moving it before the switch statement?

Reply:
It would be a valid check for both `url` and `title` actions :+1:  I'll move it to the top of the method and add a few tests for `title` type actions too.